test/suite/test_excl.py

```diff
@@ -26,23 +26,40 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 
+from helper_tiered import TieredConfigMixin, tiered_storage_sources
 import wiredtiger, wttest
 from wtscenario import make_scenarios
 
 # Test session.create with the exclusive configuration.
-class test_create_excl(wttest.WiredTigerTestCase):
-    scenarios = make_scenarios([
-        ('file', dict(type='file:')),
-        ('table', dict(type='table:'))
-    ])
-
-    # Create the object with "exclusive", then assert that creation with
-    # "exclusive" fails.
+class test_create_excl(TieredConfigMixin, wttest.WiredTigerTestCase):
+    types = [
+        ('file', dict(type = 'file:')),
+        ('table', dict(type = 'table:')),
+    ]
+
+    scenarios = make_scenarios(tiered_storage_sources, types)
+
     def test_create_excl(self):
-        uri = self.type + 'create_excl'
-        self.session.create(uri, "exclusive")
+        if self.is_tiered_scenario() and self.type == 'file:':
+            self.skipTest('Tiered storage does not support file URIs.')
+
+        uri = self.type + "create_excl"
+
+        # Create the object with the exclusive setting.
+        self.session.create(uri, "exclusive=true")
+
+        # Exclusive re-create should error.
         self.assertRaises(wiredtiger.WiredTigerError,
-            lambda: self.session.create(uri, "exclusive"))
+            lambda: self.session.create(uri, "exclusive=true"))
+
+        # Non-exclusive re-create is allowed.
+        self.session.create(uri, "exclusive=false")
+
+        # Exclusive create on a table that does not exist should succeed.
+        self.session.create(uri + "_non_existent", "exclusive=true")
```

Comment:
Isn't the `uri + _non_existent` test the same as `uri` above? Maybe we can remove `_non_existent` test?

Reply:
It's slightly different as these tables were not previously created and we're just testing the exclusive setting with the first create call. The one above test the exclusive setting after the table has already been created. But we can also remove these situations if they are not needed.

Reply:
All good, let's leave it there.